framework/src/mesh/MooseMesh.C

```diff
@@ -898,16 +898,30 @@ MooseMesh::cacheInfo()
   _subdomain_boundary_ids.clear();
   _neighbor_subdomain_boundary_ids.clear();
   _block_node_list.clear();
+  _higher_d_elem_side_to_lower_d_elem.clear();
 
   // TODO: Thread this!
   for (const auto & elem : getMesh().element_ptr_range())
   {
     SubdomainID subdomain_id = elem->subdomain_id();
+    const Elem * ip_elem = elem->interior_parent();
+
+    if (ip_elem)
+    {
+      unsigned int ip_side = ip_elem->which_side_am_i(elem);
+
+      // For some grid sequencing tests: ip_side == libMesh::invalid_uint
+      if (ip_side != libMesh::invalid_uint)
+      {
+        auto pair = std::make_pair(ip_elem, ip_side);
+        _higher_d_elem_side_to_lower_d_elem.insert(
+            std::pair<std::pair<const Elem *, unsigned short int>, const Elem *>(pair, elem));
+      }
+    }
```

Comment:
This is excellent @recuero. I was actually just wondering what additional expense constructing this mesh data structure incurred but you piggy-backed off an existing element loop. :+1: